src/coreclr/jit/ssabuilder.cpp

```diff
@@ -840,7 +854,8 @@ void SsaBuilder::RenameDef(GenTree* defNode, BasicBlock* block)
 
     GenTreeLclVarCommon* lclNode;
     bool                 isFullDef = false;
-<<<<<<< HEAD
+    ssize_t              offset    = 0;
+    unsigned             storeSize = 0;
 #if defined(TARGET_WASM)
```

Comment:
We should also set these in the `block->IsLIR()` branch (for consistency - they won't be used now as we don't expect composite definitions in late SSA):
```
if (block->IsLIR())
{
    isLocal = defNode->OperIsLocalStore();
    lclNode = isLocal ? defNode->AsLclVarCommon() : nullptr;
    if (isLocal)
    {
        isFullDef = !lclNode->IsPartialLclFld(m_pCompiler);
        offset    = lclNode->GetLclOffs();
        storeSize = isFullDef ? m_pCompiler->lvaLclExactSize(lclNode->GetLclNum()) : lclNode->AsLclFld()->GetSize();
    }
}
```

Reply:
ok, have added :
```c++
storeSize = m_pCompiler->lvaLclExactSize(lclNode->GetLclNum());

if (lclNode->OperIs(GT_STORE_LCL_FLD))
{
    offset = lclNode->AsLclFld()->GetLclOffs();
}
else
{
    offset = lclNode->GetLclOffs();
}
```

Reply:
> `storeSize = m_pCompiler->lvaLclExactSize(lclNode->GetLclNum());`

That won't be correct for `STORE_LCL_FLD`. For `offset` - `GenTreeLclVarCommon::GetLclOffs` didn't work?

Reply:
Don't fully understand the second sentence, but have changed to 
```c++
if (lclNode->OperIs(GT_STORE_LCL_FLD))
{
    offset = lclNode->AsLclFld()->GetLclOffs();
    storeSize = lclNode->AsLclFld()->GetSize();
}
```
to match `DefinesLocal` for `GT_LCL_FLD`

Reply:
> Don't fully understand the second sentence

I meant that `offset = lclNode->GetLclOffs();` should work as-is, there is no need for the `if`.

Reply:
How about:
```c++
offset = lclNode->GetLclOffs();

if (lclNode->OperIs(GT_STORE_LCL_FLD))
{
    storeSize = lclNode->AsLclFld()->GetSize();
}
else
{
    storeSize = m_pCompiler->lvaLclExactSize(lclNode->GetLclNum());
}
```

Reply:
LGTM.

Reply:
oops, didn't actually commit this, have now.